src/providers/formattingProvider.ts

```diff
@@ -164,11 +176,49 @@ export const internalFormat = (
 
         // Stop directives for formatter
         if (emptyLine) {
-            if (comment.match(/;\s*@AHK\+\+FormatBlockCommentOff/i)) {
+            if (comment.match(/;\s*@AHK\+\+AlignAssignmentOff/i)) {
+                alignAssignment = false;
+                assignmentBlock =
+                    CodeUtil.alignTextAssignOperator(assignmentBlock);
+                // Save aligned block
+                assignmentBlock.forEach((alignedFormattedLine, index) => {
+                    formattedString += buildIndentedLine(
+                        // return 'lineIndex' before 'assignmentBlock' and increment with 'index'
+                        lineIndex - assignmentBlock.length + index + 1,
+                        lines.length,
+                        alignedFormattedLine,
+                        depth,
+                        options,
+                    );
+                });
+                assignmentBlock = [];
+            } else if (comment.match(/;\s*@AHK\+\+FormatBlockCommentOff/i)) {
                 formatBlockComment = false;
             }
         }
 
+        // Align Assignment:
+        if (alignAssignment) {
+            assignmentBlock.push(formattedLine);
+            if (lineIndex !== lines.length - 1) {
+                // skip to the next iteration
+                return;
+            } else {
+                // Save aligned block if we reach end of text, but didn't find stop directive ';@AHK++AlignAssignmentOff'
+                assignmentBlock.forEach((alignedFormattedLine, index) => {
+                    formattedString += buildIndentedLine(
+                        // return 'lineIndex' before 'assignmentBlock' and increment with 'index'
+                        lineIndex - assignmentBlock.length + index + 1,
+                        lines.length,
+                        alignedFormattedLine,
+                        depth,
+                        options,
+                    );
+                });
+                assignmentBlock = [];
+            }
```

Comment:
Remove unnecessary `else` and now we have a whole 'nother indentation level saved :)

```suggestion
            }
            // Save aligned block if we reach end of text, but didn't find stop directive ';@AHK++AlignAssignmentOff'
            assignmentBlock.forEach((alignedFormattedLine, index) => {
                formattedString += buildIndentedLine(
                    // return 'lineIndex' before 'assignmentBlock' and increment with 'index'
                    lineIndex - assignmentBlock.length + index + 1,
                    lines.length,
                    alignedFormattedLine,
                    depth,
                    options,
                );
            });
            assignmentBlock = [];
```